src/ralph/admin/static/auto-complete-helpers.js

```diff
@@ -0,0 +1,73 @@
+function getCallerNode(win) {
+    var name = windowname_to_id(win.name);
+    var elem = document.querySelector('#' + name);
+    return elem;
+}
+
+function dismissRelatedLookupPopup(win, chosenId) {
+    debugger;
+    var name = windowname_to_id(win.name);
+    var elem = document.getElementById(name);
+    if (elem.className.indexOf('vManyToManyRawIdAdminField') != -1 && elem.value) {
+        elem.value += ',' + chosenId;
+    } else if (elem.nodeName.toUpperCase() === 'AUTO-COMPLETE') {
+        elem.updateById(chosenId);
+    } else {
+        document.getElementById(name).value = chosenId;
+    }
+    win.close();
+}
+
+function dismissChangeRelatedObjectPopup(win, objId, newRepr, newId) {
+    debugger;
+    elem = getCallerNode(win);
+    elem.reloadBadges();
+    win.close();
+}
+
+function dismissAddRelatedObjectPopup(win, newId, newRepr) {
+    debugger;
+    // newId and newRepr are expected to have previously been escaped by
+    // django.utils.html.escape.
+    newId = html_unescape(newId);
+    newRepr = html_unescape(newRepr);
+    var name = windowname_to_id(win.name);
+    var elem = document.getElementById(name);
+    var o;
+    if (elem.nodeName.toUpperCase() === 'AUTO-COMPLETE') {
+        elem.updateById(newId);
+    } else if (elem) {
+        var elemName = elem.nodeName.toUpperCase();
+        if (elemName == 'SELECT') {
+            o = new Option(newRepr, newId);
+            elem.options[elem.options.length] = o;
+            o.selected = true;
+        } else if (elemName == 'INPUT') {
+            if (elem.className.indexOf('vManyToManyRawIdAdminField') != -1 && elem.value) {
```

Comment:
this code seems very similar to the one in `dismissRelatedLookupPopup` - could you extract it to one function?